discord_bot.py

```diff
@@ -229,15 +216,38 @@ async def create_thread(self, parent_channel_id: int, title: str, author_id: int
     # Methods for MessageHandler protocol
     #
 
-    async def send_message(self, channel_id, message: str, file=None):
+    async def send_message(self, channel_id, message: str, file=None) -> int:
+        channel = self.get_channel(channel_id)
+        curr_message = None
         if file is not None:
             file = discord.File(file)
 
         for block in parse_blocks(message):
-            await self.get_channel(channel_id).send(block)
+            curr_message = await channel.send(block)
 
         if file is not None:
-            await self.get_channel(channel_id).send("", file=file)
+            curr_message = await channel.send("", file=file)
+
+        return curr_message.id
+
+    async def edit_message(self, channel_id: int, message_id: int, new_content: str):
+        channel = self.get_channel(channel_id)
+        try:
+            msg = await channel.fetch_message(message_id)
+            await msg.edit(content=new_content)
+        except Exception as e:
+            logging.exception(f"Could not edit message {message_id} in channel {channel_id}: {e}")
+
+    async def notify_admins(self):
```

Comment:
What if we merge the ErrorHandler and MessageHandler protocols and have a single method :`report_error` that takes the error message and a boolean `notify_admins`.

This method would report the error to the admin channels, and if `notify_admins` is true, includes the admin mention in the message.